SU2_CFD/src/solvers/CSolver.cpp

```diff
@@ -818,29 +767,13 @@ void CSolver::InitiatePeriodicComms(CGeometry *geometry,
             for (iVar = 0; iVar < nPrimVarGrad; iVar++) {
               for (iDim = 0; iDim < nDim; iDim++){
                 jacBlock[iVar][iDim] = base_nodes->GetGradient_Primitive(iPoint, iVar, iDim);
-                rotBlock[iVar][iDim] = base_nodes->GetGradient_Primitive(iPoint, iVar, iDim);
               }
             }
 
             /*--- Rotate the partial gradients in space for all variables. ---*/
 
-            for (iVar = 0; iVar < nPrimVarGrad; iVar++) {
-              if (nDim == 2) {
-                rotBlock[iVar][0] = (rotMatrix[0][0]*jacBlock[iVar][0] +
-                                     rotMatrix[0][1]*jacBlock[iVar][1]);
-                rotBlock[iVar][1] = (rotMatrix[1][0]*jacBlock[iVar][0] +
-                                     rotMatrix[1][1]*jacBlock[iVar][1]);
-              } else {
-                rotBlock[iVar][0] = (rotMatrix[0][0]*jacBlock[iVar][0] +
-                                     rotMatrix[0][1]*jacBlock[iVar][1] +
-                                     rotMatrix[0][2]*jacBlock[iVar][2]);
-                rotBlock[iVar][1] = (rotMatrix[1][0]*jacBlock[iVar][0] +
-                                     rotMatrix[1][1]*jacBlock[iVar][1] +
-                                     rotMatrix[1][2]*jacBlock[iVar][2]);
-                rotBlock[iVar][2] = (rotMatrix[2][0]*jacBlock[iVar][0] +
-                                     rotMatrix[2][1]*jacBlock[iVar][1] +
-                                     rotMatrix[2][2]*jacBlock[iVar][2]);
-              }
+            for (iVar = 0; iVar < nVar; iVar++) {
+              Rotate(zeros, jacBlock[iVar], rotBlock[iVar]);
```

Comment:
👍 much safer with Rotate()

Reply:
Much safer indeed :) :baby_chick: 

Unfortunately the loop here needs to go to nPrimVarGrad. This is the origin of the error we talked about in the SU2 dev meeting yesterday.

Reply:
![ironicPrequelMemes](https://user-images.githubusercontent.com/31306376/85033292-146ff300-b181-11ea-8cf6-80fc13879659.jpg)
... in order to keep the good tradition of prequel memes :)